### PR TITLE
feat: update to ATLAS-based FTF seeding

### PR DESCRIPTION
**Pull request by Rosie-Hasan**
Update to FTF seeding files. Now creating seeds and efficiency able to be measured 

This code doesn't include updating the code to modern C++ and renaming classes, these will be in seperate PRs

### CODE REVIEW

File: Core/include/Acts/Seeding/GNN_TrackingFilter.hpp [hunk 1]
```diff
@@ -229,8 +229,8 @@ class TrigFTF_GNN_TrackingFilter {
     const float weight_x = 0.5;
     const float weight_y = 0.5;
 
-    const float maxDChi2_x = 60.0;  // 35.0;
-    const float maxDChi2_y = 60.0;  // 31.0;
+    const float maxDChi2_x = 60.0;  // was 35.0;
+    const float maxDChi2_y = 60.0;  // was31.0;
```
Comment (LuisFelipeCoelho) [inline Core/include/Acts/Seeding/GNN_TrackingFilter.hpp:233]:
```suggestion
    const float maxDChi2_y = 60.0;  // was 31.0;
```

Comment (LuisFelipeCoelho) [reply Core/include/Acts/Seeding/GNN_TrackingFilter.hpp]: Sorry, I just realised this is still a draft

Comment (Rosie-Hasan) [reply Core/include/Acts/Seeding/GNN_TrackingFilter.hpp]: yes sorry still need to fix some formatting, thank you for pointing this one out!

**Comment by Rosie-Hasan** [comment]
This code doesn't include updating the code to modern C++ and renaming classes, these will be in seperate PRs

File: Core/include/Acts/Seeding/GNN_TrackingFilter.hpp [hunk 2]
```diff
@@ -317,7 +316,7 @@ class TrigFTF_GNN_TrackingFilter {
 
     int type = getLayerType(pS->m_n1->m_sp_FTF.combined_ID);
 
-    if (type == 0) {  // barrel TO-DO: split into barrel Pixel and barrel SCT
+    if (type == 0) {  // barrel TO-DO:c split into barrel Pixel and barrel SCT
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/GNN_TrackingFilter.hpp:319]:
```suggestion
    if (type == 0) {  // barrel TODO: split into barrel Pixel and barrel SCT
```

File: Core/include/Acts/Seeding/GNN_TrackingFilter.hpp [hunk 3]
```diff
@@ -371,7 +370,7 @@ class TrigFTF_GNN_TrackingFilter {
     });  // iterator to vector member with this id
     int index = std::distance(m_geo.begin(), iterator);
 
-    return m_geo.at(index).m_type;
+    return m_geo.at(index).m_type;  // needs to be 0,2 or -2
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/GNN_TrackingFilter.hpp:373]:
```suggestion
    return m_geo.at(index).m_type;  // needs to be 0, 2, or -2
```

File: Core/src/TrackFinding/CMakeLists.txt [hunk 4]
```diff
@@ -6,4 +6,4 @@ target_sources(
     FasTrackConnector.cpp
     RoiDescriptor.cpp
     AmbiguityTrackClustering.cpp
-)
+)
```
Comment (AJPfleger) [inline Core/src/TrackFinding/CMakeLists.txt:9]:
```suggestion
)
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]

File: Core/include/Acts/Seeding/SeedFinderFTFConfig.hpp [hunk 5]
```diff
@@ -57,12 +57,24 @@ struct SeedFinderFTFConfig {
...
+  float cut_dcurv_max = 0.001;
+  float cut_tau_ratio_max = 0.007;
+  float maxOuterRadius = 550.0;
+  float m_PtMin = 1000.0;
+  float m_tripletPtMinFrac = 0.3;
+  float m_tripletPtMin =
+      m_PtMin *
+      m_tripletPtMinFrac;  // used to be 2500, changing this helps low pt a lot
+  double ptCoeff = 0.29997 * 1.9972 / 2.0;  // from athena
+
```
Comment (LuisFelipeCoelho) [inline Core/include/Acts/Seeding/SeedFinderFTFConfig.hpp:77]: Thank you for this PR! could you add some comments here in the config file explaining what are all of these variables?

File: Examples/Scripts/Python/full_chain_itk_FTF.py [hunk 6]
```diff
@@ -78,7 +78,9 @@
     field,
     digiConfigFile=geo_dir
     / "itk-hgtd/itk-smearing-config.json",  # change this file to make it do digitization
-    outputDirRoot=outputDir,
+    # digiConfigFile=geo_dir
+    # / "itk-hgtd/itk-pixel-digitization.json",
+    # outputDirRoot=outputDir,
```
Comment (LuisFelipeCoelho) [inline Examples/Scripts/Python/full_chain_itk_FTF.py:83]: are these changes in `full_chain_itk_FTF.py` intended to be merged?

Comment (AJPfleger) [reply Examples/Scripts/Python/full_chain_itk_FTF.py:83]: good catch :)

File: Examples/Scripts/Python/full_chain_itk_FTF.py [hunk 7]
```diff
@@ -52,7 +52,7 @@
             mean=acts.Vector4(0, 0, 0, 0),
         ),
         rnd=rnd,
-        outputDirRoot=outputDir,
+        # outputDirRoot=outputDir,
```
Comment (LuisFelipeCoelho) [inline Examples/Scripts/Python/full_chain_itk_FTF.py:55]: same here and in `events=1000`. Are these intende to be merged?

**Review by LuisFelipeCoelho — CHANGES_REQUESTED** [review]

File: Examples/Scripts/Python/full_chain_itk_FTF.py (hunk 7 above)
Comment (Rosie-Hasan) [reply Examples/Scripts/Python/full_chain_itk_FTF.py]: no sorry, thank you for catching this, will change now

File: Examples/Scripts/Python/full_chain_itk_FTF.py (hunk 6 above)
Comment (Rosie-Hasan) [reply Examples/Scripts/Python/full_chain_itk_FTF.py]: No sorry, thank you for catching this!

File: Core/include/Acts/Seeding/SeedFinderFTFConfig.hpp (hunk 5 above)
Comment (Rosie-Hasan) [reply Core/include/Acts/Seeding/SeedFinderFTFConfig.hpp]: Thank you for reviewing! I have added in descriptions

File: Core/src/TrackFinding/CMakeLists.txt (hunk 4 above)
Comment (Rosie-Hasan) [reply Core/src/TrackFinding/CMakeLists.txt:9]: Thank you for reveiwing, sorry this is not letting me commit the change as it says there is no change to the code

**Review by AJPfleger — DISMISSED** [review]
lgtm now :)

**Review by LuisFelipeCoelho — APPROVED** [review]
Thank you for adding the comments. It also looks go to me

**Review by AJPfleger — APPROVED** [review]

**Comment by Rosie-Hasan** [comment]
Thank you for your help, I think the only thing still failing is the milestone check, sorry I am not sure how to add a milestone or if this is something I have permision to do

**Comment by paulgessinger** [comment]
Didn't see this before, sorry!
Let's get this in!

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [07e829dcd0713d4e000f1c687f9006e4f33c4141]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34477648) |
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476770) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476769) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476768) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476767) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476766) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476765) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476764) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476763) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476762) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476761) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476759) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476758) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476757) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476756) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476755) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476754) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476753) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476752) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476751) |
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476750) |
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34476749) |

**Pull request merged**